pybedlite/overlap_detector.py

```diff
@@ -322,18 +322,38 @@ def overlaps_any(self, interval: Span) -> bool:
             True if and only if the given interval overlaps with any interval in this
             detector.
         """
-        tree = self._refname_to_tree.get(interval.refname)
+        tree = self._refname_to_tree.get(interval.refname, None)
         if tree is None:
             return False
         else:
             if not self._refname_to_indexed[interval.refname]:
                 tree.index()
-            try:
-                next(iter(tree.overlap(interval.refname, interval.start, interval.end)))
-            except StopIteration:
-                return False
-            else:
-                return True
+                self._refname_to_indexed[interval.refname] = True
+            # IntervalSet uses closed intervals whereas we are using half-open intervals, so add 1
+            # to start
+            return tree.any_overlaps(interval.start + 1, interval.end)
+
+    def iter_overlaps(self, interval: Span) -> Iterator[SpanType]:
```

Comment:
This new method isn't much different in behavior than `get_overlaps` so I don't think we need both publicly available. Although this one looks lazy, it isn't.

Reply:
Responding to @clintval and @msto in the same place since you made similar comments:

- It's not intended to be lazy, just unopinionated about the container and sort order. If you just want to iterate over all the overlaps in any order, this saves processing and memory, as well as keeping any duplicates (as far as `__hash__` is concernced).
- In another project I had to write my own subclass with duplicate overlap-getting logic because I explicitly wanted duplicates and couldn't get them otherwise. But I can't exactly change the behavior of `get_overlaps` without breaking the API.
- To me this design seemed like a compromise that kept all the overlap logic in one place while allowing a lighter-weight call for those who wanted it. But it's outside the remit of the originating issue, so I could punt and remove this if the consensus is that it's unwanted.

Reply:
I'd usually favor including those as options in `get_overlaps()` instead of having a separate method.

e.g. 

```py
def get_overlaps(
    self,
    interval: Span,
    sort_overlaps: bool = True,
    include_duplicates: bool = False,
) -> list[SpanType]:
```

I agree with Clint that returning an `Iterator` implies lazy evaluation. If the method _can_ be lazy, I think our usual pattern is to return an iterator and allow the user to convert to `list` at the call site. Otherwise, the method should return `list`.